_P127_Teleinfo.ino

```diff
@@ -47,250 +68,132 @@ boolean Plugin_127(byte function, struct EventStruct *event, String& string)
-        arg.toCharArray(argc, 25);
-        tmpString = WebServer.arg(argc);
-        strncpy(deviceTemplate[1], tmpString.c_str(), sizeof(deviceTemplate[1]));
-
-        Settings.TaskDeviceID[event->TaskIndex] = 1; // temp fix, needs a dummy value
-
-        SaveCustomTaskSettings(event->TaskIndex, (byte*)&deviceTemplate, sizeof(deviceTemplate));
-        success = true;
+        serialHelper_getGpioNames(event, false, false); // TX optional
         break;
       }
 
-
-    case PLUGIN_INIT: // ok
+    case PLUGIN_INIT:
       {
-        char deviceTemplate[2][128];
-        LoadCustomTaskSettings(event->TaskIndex, (byte*)&deviceTemplate, sizeof(deviceTemplate));
-        P127_HOST = String(deviceTemplate[0]);
-        P127_PORT = int(Settings.TaskDevicePluginConfig[event->TaskIndex][0]);
-        P127_URL = String(deviceTemplate[1]);
-        Serial.begin(1200, SERIAL_7E1); //Liaison série avec les paramètres
+        // Init Serial
+        const int16_t serial_rx = CONFIG_PIN1;
+        const int16_t serial_tx = CONFIG_PIN2;        
+        P127_easySerial = new ESPeasySerial(serial_rx, serial_tx);
+
+        String log = F("P127 : Init ");
+        if (PCONFIG(0)){
+          // Init teleinfo and serial for HISTORIQUE mode
+          tinfo.init(true);          
+          P127_easySerial->begin((unsigned long)1200, SERIAL_7E1, SERIAL_RX_ONLY);
+          log += F("HISTORIQUE mode 1200_7E1 RX=");          
+        }else{
+          // Init teleinfo and serial for STANDARD mode
+          tinfo.init(false);          
+          P127_easySerial->begin((unsigned long)9600, SERIAL_7E1, SERIAL_RX_ONLY);
+          log += F("STANDARD mode 9600_7E1 RX=");          
+        }
+        
+        log += serial_rx;        
+        log += F(" reading ");       
+        log += ExtraTaskSettings.TaskDeviceValueNames[0];
+        log += F(",");       
+        log += ExtraTaskSettings.TaskDeviceValueNames[1];
+        addLog(LOG_LEVEL_INFO, log);
+        
         Plugin_127_init = true;
         success = true;
         break;
       }
 
-
-    case PLUGIN_SERIAL_IN:
+    case PLUGIN_READ:
       {
-        uint8_t serial_buf[P127_BUFFER_SIZE];
-        int RXWait = 10;
-        int timeOut = RXWait;
-        size_t bytes_read = 0;
-        char log[40];
-
+        if ( ! Plugin_127_init ) break ;
 
-        while (timeOut > 0)
-        {
-          while (Serial.available()) {
-            if (bytes_read < P127_BUFFER_SIZE) {
-              serial_buf[bytes_read] = Serial.read();
-              bytes_read++;
-            }
-            else {
-              Serial.read();  // when the buffer is full, just read remaining input, but do not store...
-
-            }
-            timeOut = RXWait; // if serial received, reset timeout counter
-          }
-          delay(1);
-          timeOut--;
-        }
+        event->sensorType = SENSOR_TYPE_DUAL;
+        success = true;
+        int value;
+        char cvalue[TAILLE_MAX_VALUE]; 
```

Comment:
Where is `TAILLE_MAX_VALUE` defined?
And what value has it? (you're declaring on the stack here, so it should not be too big)

Reply:
TAILLE_MAX_VALUE comes from https://github.com/marco402/LibTeleinfo/blob/rewrite-Wifinfo/examples/Wifinfo/LibTeleinfo.h#L102
and TAILLE_MAX_VALUE =  98

Reply:
If external files are needed to make it compile, then you should include them, or at least post a link to them in comments at the include.

Otherwise it is next to impossible for others to use the code.

Reply:
code included in P127_LibTeleinfo_Library directory with minimal Readme.md info